### DISCUSSION TITLE
format of customKcats.tsv with and/or relations

### DISCUSSION THREAD

**Discussion by HossFir** [SysBioChalmers/GECKO#364] [GECKO3]
I have GPR Associations for PSII Reaction (it will split into six reactions for the ecModel.):

After some attempts, I received this warning:
WARNING: No matches found. Consider checking the IDs or proteins in customKcats.

**PSII GPR's:**

(sll1867 and sll0849 and slr0906 and sll0851 and ssr3451 and smr0006 and ssl2598 and sml0001 and smr0008 and sml0005 and smr0007 and sml0003 and smr0009 and sll0427 and sll1418 and sll1194 and sll0258 and sll1398 and slr1739 and sml0002 and sml0007 and slr1645 and sll1578 and sll1577 and sll1579 and sll1580 and ssl3093 and slr1878 and sll1051 and slr2051 and slr2067 and slr1986 and ssr3383 and sll0928 and slr0335 and slr1459) **OR** (sll1867 and slr0927 and slr0906 and sll0851 and ssr3451 and smr0006 and ssl2598 and sml0001 and smr0008 and sml0005 and smr0007 and sml0003 and smr0009 and sll0427 and sll1418 and sll1194 and sll0258 and sll1398 and slr1739 and sml0002 and sml0007 and slr1645 and sll1578 and sll1577 and sll1579 and sll1580 and ssl3093 and slr1878 and sll1051 and slr2051 and slr2067 and slr1986 and ssr3383 and sll0928 and slr0335 and slr1459) **OR** (slr1181 and sll0849 and slr0906 and sll0851 and ssr3451 and smr0006 and ssl2598 and sml0001 and smr0008 and sml0005 and smr0007 and sml0003 and smr0009 and sll0427 and sll1418 and sll1194 and sll0258 and sll1398 and slr1739 and sml0002 and sml0007 and slr1645 and sll1578 and sll1577 and sll1579 and sll1580 and ssl3093 and slr1878 and sll1051 and slr2051 and slr2067 and slr1986 and ssr3383 and sll0928 and slr0335 and slr1459) **OR** (slr1181 and slr0927 and slr0906 and sll0851 and ssr3451 and smr0006 and ssl2598 and sml0001 and smr0008 and sml0005 and smr0007 and sml0003 and smr0009 and sll0427 and sll1418 and sll1194 and sll0258 and sll1398 and slr1739 and sml0002 and sml0007 and slr1645 and sll1578 and sll1577 and sll1579 and sll1580 and ssl3093 and slr1878 and sll1051 and slr2051 and slr2067 and slr1986 and ssr3383 and sll0928 and slr0335 and slr1459) **OR** (slr1311 and sll0849 and slr0906 and sll0851 and ssr3451 and smr0006 and ssl2598 and sml0001 and smr0008 and sml0005 and smr0007 and sml0003 and smr0009 and sll0427 and sll1418 and sll1194 and sll0258 and sll1398 and slr1739 and sml0002 and sml0007 and slr1645 and sll1578 and sll1577 and sll1579 and sll1580 and ssl3093 and slr1878 and sll1051 and slr2051 and slr2067 and slr1986 and ssr3383 and sll0928 and slr0335 and slr1459) **OR** (slr1311 and slr0927 and slr0906 and sll0851 and ssr3451 and smr0006 and ssl2598 and sml0001 and smr0008 and sml0005 and smr0007 and sml0003 and smr0009 and sll0427 and sll1418 and sll1194 and sll0258 and sll1398 and slr1739 and sml0002 and sml0007 and slr1645 and sll1578 and sll1577 and sll1579 and sll1580 and ssl3093 and slr1878 and sll1051 and slr2051 and slr2067 and slr1986 and ssr3383 and sll0928 and slr0335 and slr1459)

**Which has the corresponding UniProt IDs of:**

P16033 + P09192 + P05429 + P09193 + P09190 + P09191 + P14835 + Q54697 + P73070 + P15819 + Q55354 + P72701 + P26286 + P10549 + P73952 + Q55332 + Q55013 + Q55356 + P73731 + P72575 + P73676 + P74367 + Q54715 + Q54714 + P73204 + P73203 + P73202 + P73638 + P72652 + P73093 + Q01951 + Q01952 + Q01950 + P72870 + Q55544 + P74551 **OR** P16033 + P09192 + P05429 + P09193 + P09190 + P09191 + P14835 + Q54697 + P73070 + P15819 + Q55354 + P72701 + P26286 + P10549 + P73952 + Q55332 + Q55013 + Q55356 + P73731 + P72575 + P73676 + P74367 + Q54715 + Q54714 + P73204 + P73203 + P73202 + P73638 + P72652 + P73093 + Q01951 + Q01952 + Q01950 + P72870 + Q55544 + P74551 **OR** P07826 + P09192 + P05429 + P09193 + P09190 + P09191 + P14835 + Q54697 + P73070 + P15819 + Q55354 + P72701 + P26286 + P10549 + P73952 + Q55332 + Q55013 + Q55356 + P73731 + P72575 + P73676 + P74367 + Q54715 + Q54714 + P73204 + P73203 + P73202 + P73638 + P72652 + P73093 + Q01951 + Q01952 + Q01950 + P72870 + Q55544 + P74551 **OR** P07826 + P09192 + P05429 + P09193 + P09190 + P09191 + P14835 + Q54697 + P73070 + P15819 + Q55354 + P72701 + P26286 + P10549 + P73952 + Q55332 + Q55013 + Q55356 + P73731 + P72575 + P73676 + P74367 + Q54715 + Q54714 + P73204 + P73203 + P73202 + P73638 + P72652 + P73093 + Q01951 + Q01952 + Q01950 + P72870 + Q55544 + P74551 **OR** P16033 + P09192 + P05429 + P09193 + P09190 + P09191 + P14835 + Q54697 + P73070 + P15819 + Q55354 + P72701 + P26286 + P10549 + P73952 + Q55332 + Q55013 + Q55356 + P73731 + P72575 + P73676 + P74367 + Q54715 + Q54714 + P73204 + P73203 + P73202 + P73638 + P72652 + P73093 + Q01951 + Q01952 + Q01950 + P72870 + Q55544 + P74551 **OR** P16033 + P09192 + P05429 + P09193 + P09190 + P09191 + P14835 + Q54697 + P73070 + P15819 + Q55354 + P72701 + P26286 + P10549 + P73952 + Q55332 + Q55013 + Q55356 + P73731 + P72575 + P73676 + P74367 + Q54715 + Q54714 + P73204 + P73203 + P73202 + P73638 + P72652 + P73093 + Q01951 + Q01952 + Q01950 + P72870 + Q55544 + P74551

I also want to update stoichiometry column in the CustomKcats.tsv, so I consider protein copies in the complex (so that the stoicho column will be sth like this as an example: 2 + 2 + 10 + 1 + 10 + 9 + 4 + 5 + 7 + 9 + 3 + 2 + 5 + 10 + 6 + 6 + 10 + 5 + 4 + 3 + 1 + 9 + 10 + 1 + 8 + 7 + 3 + 4 + 5 + 3 + 1 + 3 + 10 + 1 + 4 + 7)

I wonder how I should arrange this into the CustomKcats.tsv for the corresponding columns of proteins, kcat, rxn, and stochio?

**Comment by edkerk** [comment]
You should write the six GPRs on separate lines. Pay attention that there are also empty columns that should still be provided. Example with shortened GPRs:

| proteins                   | genes                                   | gene_name                         | kcat | rxns | notes     | stoicho          |
|----------------------------|-----------------------------------------|-----------------------------------|------|------|-----------|------------------|
| P16033 + P09192 + P05429 + | sll1867 and slr0927 and slr0906 and ... | gene1 and gene2 and gene3 and ... | 61   |      | customTxt | 2 + 2 + 10 + ... |
| P16033 + P09192 + P05429 + | sll1867 and sll0849 and slr0906 and ... | gene1 and gene2 and gene3 and ... | 15   |      | customTxt | 2 + 2 + 10 + ... |
|                            |                                         |                                   |      |      |           |                  |

The columns are tab-separated.

Note that the `genes` and `gene_name` fields are not actually used, they are provided as reference information in the `customKcats.tsv` and provide legacy support. The order in `stoicho` should match the order of the proteins in `proteins`.
